### PR TITLE
add metadata extractors

### PR DESCRIPTION
**Pull request by schosterbarak**
The goal is to automatically extract CI metadata when checkov runs in that context

### CODE REVIEW

File: checkov/common/bridgecrew/run_metadata/abstract_run_metadata_extractor.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    pr_id = ""
+    pr_url = ""
+    commit_hash = ""
+    commit_url = ""
+    author_name = ""
+    author_url = ""
+    run_id = ""
+    run_url = ""
+    repository_url = ""
+
```
Comment (gruebel) [inline checkov/common/bridgecrew/run_metadata/abstract_run_metadata_extractor.py:30]: please remove this, class level attributes are not needed here, and you don't leverage them 😄

File: checkov/common/bridgecrew/run_metadata/abstract_run_metadata_extractor.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            self.to_branch = to_branch
+            self.pr_id = pr_id
+            self.pr_url = pr_url
+            self.commit_hash = commit_hash
+            self.commit_url = commit_url
+            self.author_name = author_name
+            self.author_url = author_url
+            self.run_id = run_id
+            self.run_url = run_url
+            self.repository_url = repository_url
```
Comment (gruebel) [inline checkov/common/bridgecrew/run_metadata/abstract_run_metadata_extractor.py:53]:
```suggestion
        self.from_branch = from_branch
        self.to_branch = to_branch
        self.pr_id = pr_id
        self.pr_url = pr_url
        self.commit_hash = commit_hash
        self.commit_url = commit_url
        self.author_name = author_name
        self.author_url = author_url
        self.run_id = run_id
        self.run_url = run_url
        self.repository_url = repository_url
```
looking at the later usage, I don't see a need to do the if condition here, just set all the attributes and later you decide via `is_current_ci()` to access the correct one

File: checkov/common/bridgecrew/run_metadata/registry.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+class RunMetadataExtractorsRegistry():
+    def __init__(self):
+        self.extractors = set()
+
+    def register(self, extractor) -> None:
+        self.extractors.add(extractor)
+
+    def get_extractor(self):
+        for extractor in self.extractors:
+            if extractor.is_current_ci:
```
Comment (gruebel) [inline checkov/common/bridgecrew/run_metadata/registry.py:12]:
```suggestion
            if extractor.is_current_ci():
```
does it wok without the round brackets?

File: docs/6.Contribution/Implementing CI Metadata Extractor.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# Implementing CI Metadata extractor
+CI/CD jobs have environment variables that can enrich the execution context.
+Attributes like:
+1. Author of the run
+2. Commit sha
+3. Pull request ID
+4. Link to the host running the CI 
+
+Those attributed can be added by reading environment variables published on the public docs of CI/CD vendors.
```
Comment (gruebel) [inline docs/6.Contribution/Implementing CI Metadata Extractor.md:9]:
```suggestion
Those attributes can be added by reading environment variables published on the public docs of CI/CD vendors.
```

File: docs/6.Contribution/Implementing CI Metadata Extractor.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+Examples: 
+https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
+https://docs.github.com/en/actions/learn-github-actions/environment-variables
+
+## How to implement a new Run metadata extractor?
+You'll need to implement a new class derrived from RunMetaDataExtractor and commit it into the directory `checkov/common/bridgecrew/run_metadata/extractors`.
```
Comment (gruebel) [inline docs/6.Contribution/Implementing CI Metadata Extractor.md:15]:
```suggestion
You'll need to implement a new class derived from RunMetaDataExtractor and commit it into the directory `checkov/common/bridgecrew/run_metadata/extractors`.
```

File: checkov/common/bridgecrew/platform_integration.py [hunk 6]
```diff
@@ -453,59 +453,79 @@ def get_public_run_config(self) -> None:
...
+                                                                                                                                                                u"\u2022 " + "Integration with up to 100 cloud resources for:\n"
+                                                                                                                                                                             "\t" + u"\u25E6 " + "\tAutomated cloud resource checks\n"
+                                                                                                                                                                                                 "\t" + u"\u25E6 " + "\tResource drift detection\n"
+                                                                                                                                                                                                                     "\n"
+                                                                                                                                                                                                                     "\n" + "and much more...",
```
Comment (gruebel) [inline checkov/common/bridgecrew/platform_integration.py:478]: this looks funny 😂 need to adjust it an other time

**Review by gruebel — COMMENTED** [review]
in general looks good, I just would like not to use class attributes in `abstract_run_metadata_extractor.py`, which should be possible to do.

File: checkov/common/bridgecrew/run_metadata/extractors/bitbucket.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+                         to_branch=to_branch,
+                         pr_id=pr_id,
+                         pr_url=pr_url,
+                         commit_hash=commit_hash,
+                         commit_url=commit_url,
+                         author_name=author_name,
+                         author_url=author_url,
+                         run_id=run_id,
+                         run_url=run_url,
+                         repository_url=repository_url)
```
Comment (nimrodkor) [inline checkov/common/bridgecrew/run_metadata/extractors/bitbucket.py:35]: @schosterbarak Can we extract these variables in a method that gets called, instead on import?

**Review by nimrodkor — APPROVED** [review]
We're good :)

**Review by tronxd — APPROVED** [review]

**Pull request merged**